tests/System.IO.Abstractions.Tests/DirectoryWrapperTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using NUnit.Framework;
+
+namespace System.IO.Abstractions.Tests
+{
+    [TestFixture]
+    public class DirectoryWrapperTests
+    {
+        [Test]
+        public void GetParent_ForRootDirectory_ShouldReturnNull()
+        {
+            // Arrange
+            var wrapperFilesystem = new FileSystem();
+
+            // Act
+            var result = wrapperFilesystem.Directory.GetParent(@"C:\");
```

Comment:
This does not work on Linux because there is no `C:` drive. See e.g. https://github.com/System-IO-Abstractions/System.IO.Abstractions/pull/638/checks?check_run_id=1098272021#step:6:3572

To make this generic you could use `Directory.GetDirectoryRoot`, something like this:

```csharp
var root = wrapperFilesystem.Directory.GetDirectoryRoot(".");
var result = wrapperFilesystem.Directory.GetParent(root);
```